enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/ServiceDiscoveryEnricher.java

```diff
@@ -107,22 +118,42 @@ public ServiceDiscoveryEnricher(EnricherContext buildContext) {
 
         if (kind == Kind.SERVICE) {
 
-            tryCamelDSLProject();
+            if (!tried) {
+                tryCamelDSLProject();
+                //try more project types in future, last match wins
+                tried = true;
+            }
 
             if (discoverable != null) {
                 Map<String, String> labels = new HashMap<>();
-                String labelName = PREFIX + DISCOVERABLE;
+                String labelName = PREFIX;
```

Comment:
is it really a PREFIX still ?

Reply:
We kept the annotation prefix the same as the label name on purpose so it's obvious they go together.